### PR TITLE
Scripts for base and extractors

### PR DESCRIPTION
**Pull request by Afohtim**

### CODE REVIEW

File: Assets/Scripts/Objects/GameResource.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    metals,
+    rareMetals,
+    dust,
+    radioactive,
+
+    concrete,
+    electronics,
+    machineParts,
+    steel,
```
Comment (bfilonenko) [inline Assets/Scripts/Objects/GameResource.cs]:
```suggestion
    Food,
    Oxygen,
    Water,
    Elecricity,

    Metals,
    RareMetals,
    Dust,
    Radioactive,

    Concrete,
    Electronics,
    MachineParts,
    Steel,
```

Should be better to use Pascal Casing for names of enumeration types, as in [ .NET Design-Guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations)

File: Assets/Scripts/Objects/GameResourcesStorage.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+public enum ResourcePack
+{
+    noResources,
+    firstBaseResources
```
Comment (bfilonenko) [inline Assets/Scripts/Objects/GameResourcesStorage.cs]:
```suggestion
    NoResources,
    FirstBaseResources
```

Should be better to use Pascal Casing for names of enumeration types, as in [ .NET Design-Guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations)

File: Assets/Scripts/Objects/GameResourcesStorage.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        Debug.Assert(count >= 0);
+        Debug.Assert((int)resource < resourceCount);
+        storage[(int)resource].amount += count;
+    }
+
+    public void RemoveResource(GameResourceType resource, float count)
+    {
+        Debug.Assert(count >= 0);
+        Debug.Assert(storage[(int)resource].amount - count >= 0);
+        Debug.Assert((int)resource < 12);
```
Comment (bfilonenko) [inline Assets/Scripts/Objects/GameResourcesStorage.cs]:
```suggestion
        Debug.Assert((int)resource < resourceCount);
```

File: Assets/Scripts/Objects/InBaseMoxie.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+public class InBaseMoxie : InBaseObject
+{
+    public InBaseMoxie(Base parent, float oxygenPerSecond = 1) : base(parent)
+    {
+        this.resourcePerSecond = oxygenPerSecond;
+        resourceType = GameResourceType.oxygen;
```
Comment (bfilonenko) [inline Assets/Scripts/Objects/InBaseMoxie.cs]:
```suggestion
        resourceType = GameResourceType.Oxygen;
```

File: Assets/Scripts/Objects/InBaseVaporator.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+
+public class InBaseVaporator : InBaseObject
+{
+    public InBaseVaporator(Base parent, float waterPerSecond = 1) : base(parent) 
+    {
+        this.resourcePerSecond = waterPerSecond;
+        resourceType = GameResourceType.water;
```
Comment (bfilonenko) [inline Assets/Scripts/Objects/InBaseVaporator.cs]:
```suggestion
        resourceType = GameResourceType.Water;
```

File: Assets/Scripts/Objects/ResourceDeposit.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+public class ResourceDeposit : MonoBehaviour
+{
+    private GameResource resource;
+    private bool isResourseSet = false;
+    public int max_amount;
```
Comment (bfilonenko) [inline Assets/Scripts/Objects/ResourceDeposit.cs]:
```suggestion
    public int maxAmount;
```

**Review by bfilonenko — CHANGES_REQUESTED** [review]

**Review by SymphonyOfTranquility — APPROVED** [review]

**Review by bfilonenko — APPROVED** [review]

**Pull request merged**